### PR TITLE
Migrate functionality from TxMapAndMaybeTrees to TRBID where appropriate.

### PR DESCRIPTION
**Pull request by zancas**
Draft PR #982 builds on this PR.

I tried to note where methods could not because of their dependence on witness trees.

Methods that *have* moved are tested and shown to be dependent only on `transaction_records_by_id` (and therefore should be implented on a trbid).

### CODE REVIEW

**Review by fluidvanadium — DISMISSED** [review]
this is awesome! it had to happen at some point, the sooner the better

File: zingolib/src/wallet/transactions/recording.rs [hunk 1]
```diff
@@ -21,33 +21,12 @@ use crate::{
         traits::{self, DomainWalletExt, Nullifier, Recipient},
     },
 };
-
-use super::TxMapAndMaybeTrees;
-impl TxMapAndMaybeTrees {
```
Comment (fluidvanadium) [inline zingolib/src/wallet/transactions/recording.rs:26]: it is confusing to me that this file has impl blocks for both

**Review by fluidvanadium — CHANGES_REQUESTED** [review]
i really like what is going on here, however i feel very strongly that impl TransactionRecordsById does not go in transactions/recording.rs next to TxMapAndMaybeTrees

**Comment by zancas** [comment]
That's where the methods were implemented before.  In "recording.rs".  I'm pretty sure you put them there.

File: zingolib/src/wallet/transactions/recording.rs [hunk 2]
```diff
@@ -446,16 +293,172 @@ impl TxMapAndMaybeTrees {
...
+            }
+        } else {
+            ZingoLibError::UnknownError.handle()?
+        }) // todO add special error variant
     }
 }
 
 // shardtree
-impl TxMapAndMaybeTrees {
+impl super::TxMapAndMaybeTrees {
```
Comment (AloeareV) [inline zingolib/src/wallet/transactions/recording.rs]: I believe we're moving in favor of absolute paths over super. This is more verbose, is is clearer?
```suggestion
impl crate::wallet::transactions::TxMapAndMaybeTrees {
```

**Review by AloeareV — DISMISSED** [review]

File: zingolib/src/wallet/transactions/recording.rs [hunk 3]
```diff
@@ -21,33 +21,12 @@ use crate::{
...
-                .truncate_removing_checkpoint(&(reorg_height - 1))
-                .expect("Infallible");
-            t.witness_tree_orchard
-                .truncate_removing_checkpoint(&(reorg_height - 1))
-                .expect("Infallible");
-            t.add_checkpoint(reorg_height - 1);
-        }
-    }
-
+impl super::TransactionRecordsById {
```
Comment (fluidvanadium) [inline zingolib/src/wallet/transactions/recording.rs]: please move this impl block to either transaction_records_by_id.rs or a submod of transaction_records_by_id

**Review by fluidvanadium — CHANGES_REQUESTED** [review]

File: zingolib/src/wallet/transactions/recording.rs (hunk 3 above)
Comment (zancas) [reply zingolib/src/wallet/transactions/recording.rs]: Done.

File: zingolib/src/wallet/transactions/recording.rs [hunk 4]
```diff
@@ -42,67 +35,9 @@ impl TxMapAndMaybeTrees {
         }
     }
 
-    /// Invalidates all those transactions which were broadcast but never 'confirmed' accepted by a miner.
-    pub(crate) fn clear_expired_mempool(&mut self, latest_height: u64) {
```
Comment (fluidvanadium) [inline zingolib/src/wallet/transactions/recording.rs:46]: i have confirmed that this function does not effect witness trees

**Review by fluidvanadium — APPROVED** [review]
amazing!

**Pull request merged**